### DISCUSSION TITLE
How do I checkout a branch??

### DISCUSSION THREAD

**Discussion by beardwin** [NeogitOrg/neogit#1591] [Q&A]
Neogit newb here. I've read through everything and I'm either missing it or I'm just... dumb.

If I do the following:
> :Neogit branch
> b //to checkout a branch

I then get a list of branches. If I <C-n> or type the name so that the branch is hIghlighted, I would expect hitting <cr> would checkout the branch. Instead, the popup just closes with nothing having happened. I've tried slapping `Tab`, which autofills the branch name, but then hitting enter still doesn't switch branches. Just exits the popup.

What am I doing wrong?

Thanks!

**Comment by beardwin** [comment]
welps.... after an hour it just started working and I have no idea why. I didn't change a thing.

> it just dawned on me.... I likely had uncommitted changes. Though I would have expected an error or something. Maybe because I'm using Notify? not sure.

**Comment by CKolkey** [comment]
Weird 🤷

FWIW I usually use "l" for a local branch instead of "b" for any revision because if you select a remote ref with "b" it will detach your head, but will create a local branch tracking the remote with "l"

**Reply by beardwin** [reply]
Yea that's a good tip. Thanks dude. 

I will try and figure out why I wasn't receiving any kind of warning re: uncommitted changes when trying to switch branches. My gut says it has something to do with Notify.nvim obscuring it (just setup Noice and Neogit last night and don't really have a good lay of the land yet).
